lib/ansible/modules/cloud/amazon/elasticache_subnet_group.py

```diff
@@ -98,43 +97,48 @@ def main():
                 module.fail_json(msg=str("Parameter %s not allowed for state='absent'" % not_allowed))
 
     # Retrieve any AWS settings from the environment.
-    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
 
     if not region:
         module.fail_json(msg=str("Either region or AWS_REGION or EC2_REGION environment variable or boto config aws_region or ec2_region must be set."))
 
     """Get an elasticache connection"""
-    try:
-        conn = connect_to_region(region_name=region, **aws_connect_kwargs)
-    except boto.exception.NoAuthHandlerFound as e:
-        module.fail_json(msg=e.message)
+
+    connection = boto3_conn(module, conn_type='client',
+                            resource='elasticache', region=region,
+                            endpoint=ec2_url, **aws_connect_kwargs)
 
     try:
         changed = False
         exists = False
 
         try:
-            matching_groups = conn.describe_cache_subnet_groups(group_name, max_records=100)
+            matching_groups = connection.describe_cache_subnet_groups(
+                CacheSubnetGroupName=group_name, MaxRecords=100)
             exists = len(matching_groups) > 0
-        except BotoServerError as e:
-            if e.error_code != 'CacheSubnetGroupNotFoundFault':
-                module.fail_json(msg=e.error_message)
+        except botocore.exceptions.ClientError as e:
```

Comment:
You should handle botocore.exceptions.BotoCoreError as well as ClientError. fail_json_aws and is_boto3_error_code will help simplify this.